src/main/java/com/codejoust/main/service/GameManagementService.java

```diff
@@ -283,14 +307,131 @@ public GameDto manuallyEndGame(String roomId, EndGameRequest request) {
         return gameDto;
     }
 
-    protected void handleEndGame(Game game) {
+    public void handleEndGame(Game game) {
         // Cancel all previously scheduled timers
         GameTimer gameTimer = game.getGameTimer();
         gameTimer.getTimer().cancel();
 
         for (Timer timer : gameTimer.getNotificationTimers()) {
             timer.cancel();
         }
+
+        CreateGameReportTask createGameReportTask = new CreateGameReportTask(this, game);
+        Timer createGameReportTimer = new Timer();
+        createGameReportTimer.schedule(createGameReportTask, GameTimer.DURATION_1 * 1000);
+    }
+
+    public GameReport createGameReport(Game game) {
```

Comment:
nit: move this + other report logic into a new ReportService

Reply:
Sounds good - done! ✅

Reply:
When I created the new `ReportService` class, I also created the new `ReportServiceTests` class; in the original tests, I also used a `Spy` for `GameManagementService` not just for the `createGameReport` method, but also to add submissions and add the game from the room. Is it ok that I have `Spy` objects for both `GameManagementService` and `ReportService` within that file, or should I find some workaround?

(I ask because this change makes the `ReportServiceTests` dependent not only on `ReportService`, but also on the `GameManagementService` - so it's not entirely isolated, but the work to get there may be large.) Thoughts @alankbi?

Reply:
Does this cause any issues? Never tried that before, but I guess if it works then it hopefully shouldn't be an issue

Reply:
It doesn't cause any issues, just makes the code styling arguably a bit worse. In the interest of keeping progress moving given how long this PR has been out, I'll merge this change in though, if that's alright. 👍